site-graph/http/error.go

```diff
@@ -0,0 +1,35 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+var (
+	ErrUnauthorized = fmt.Errorf("not authorized")
+)
+
+type MyError struct {
+	HTTPStatus int    `json:"-"`
+	Message    string `json:"message"`
+}
+
+func HttpError(w http.ResponseWriter, code int, err error) error {
+	e := &MyError{
+		HTTPStatus: code,
+		Message:    err.Error(),
+	}
+
+	b, err := json.Marshal(e)
```

Comment:
you don't need intermediate byte slice here - better 
`json.NewEncoder(w).Encode(e)`